File: problem2.js
var fibonacciNumbers = [1,2];
var max = 4000000;
var total = 0;

function GetNextFibonacci()

{
	var nextFibonacci = fibonacciNumbers[fibonacciNumbers.length-1] + fibonacciNumbers[fibonacciNumbers.length-2];
	fibonacciNumbers.push(nextFibonacci);
	return nextFibonacci;

}

while (GetNextFibonacci() <=max)
{	
	
		GetNextFibonacci();
	
}

for(i =0; i < fibonacciNumbers.length; i++)
{
	var current = fibonacciNumbers[i];
	if(current<max && current%2 == 0)
	{
		total += current;
	}

}


console.log(fibonacciNumbers);
console.log(total);

File: problem2.rb
fibonacciNumbers = [1,2]
nextFibonacci = 0
max = 4000000
total = 0


while nextFibonacci <=max
	nextFibonacci = fibonacciNumbers[fibonacciNumbers.length-1] + fibonacciNumbers[fibonacciNumbers.length-2]
	fibonacciNumbers.push(nextFibonacci)
end

i = 0

while i < fibonacciNumbers.length
	
	current = fibonacciNumbers[i]
	
	if(current<max and current%2 == 0)
		total = total + current
	end
	
	i += 1

end



puts total